core/dbt/include/global_project/macros/materializations/models/incremental/strategies.sql

```diff
@@ -0,0 +1,83 @@
+{% macro get_incremental_append_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_append_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+
+{% macro default__get_incremental_append_sql(arg_dict) %}
+
+  {% do return(get_insert_into_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{# snowflake #}
+{% macro get_incremental_delete_insert_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_delete_insert_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_delete_insert_sql(arg_dict) %}
+
+  {% do return(get_delete_insert_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["unique_key"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{# snowflake, bigquery, spark #}
+{% macro get_incremental_merge_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_merge_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_merge_sql(arg_dict) %}
+
+  {% do return(get_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["unique_key"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
```

Comment:
I'm not sure exactly how we want to factor this going forward when add new capabilities to these macros, leveraging the extensibility of `arg_dict`. E.g. If we wanted to support `predicates` as a user-provided config for the `merge` strategy, which templates out additional conditions for the resulting `MERGE` statement —  would we modify the signature of `get_merge_sql`? Pass in optional `**kwargs`, so that it works whether or not the `predicates` argument is provided?

Reply:
I've been wondering about this too. It almost feels easier for people to add new extra config that can be pulled out in the user incremental strategy. Because the creation of that arg_dict happens in the materialization macro, so new arguments would require modifying the materialization macro, which seems like the wrong direction.  I did include predicates (from an extra 'incremental_predicates' config) in the arg_dict.

Reply:
We could do a "by convention" thing where we pull in all configs that start with 'incremental_strategy' or something. It does feel a bit hacky, but I haven't come up with a solution that's better yet.

Reply:
Or have an incremental_strategy dictionary with extra args